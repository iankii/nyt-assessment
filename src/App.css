.header {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: gray;
  padding: 16px;
}