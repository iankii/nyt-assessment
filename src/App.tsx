import React from "react";
import { Outlet } from "react-router-dom";
import { Header, Footer } from "./components/index";

function App() {
  return (
    <div className="min-h-screen flex-wrap content-between bg-light">
      <div className="w-full block">
        <Header />
        <main>
          <Outlet />
        </main>
        <Footer />
      </div>
    </div>
  );
}

export default App;
