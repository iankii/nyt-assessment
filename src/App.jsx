import React from "react";
import "./App.css";
import { Link, Outlet } from "react-router-dom";

function App() {
	return (
		<div className="app w-full" data-testid="app">
			{/* <Header /> */}
			<header className="header" data-testid="header">
				Some Header {"   "}
				<Link to="/">Home</Link>
			</header>
			<main>
				<Outlet />
			</main>
			<footer>Some footer</footer>
			{/* <Footer /> */}
		</div>
	);
}

export default App;
