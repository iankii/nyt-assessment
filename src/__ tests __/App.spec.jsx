import React from "react";
import App from "../App";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

describe("Renders the main page", () => {
	beforeAll(() => {
		render(
			<MemoryRouter>
				<App />
			</MemoryRouter>
		);
	});

	it("should render App component", () => {
		const app = screen.getByTestId("app");
		const h1 = screen.getByTestId("header");

		expect(app).toBeInTheDocument();
		expect(h1).toBeInTheDocument();
	});
});
