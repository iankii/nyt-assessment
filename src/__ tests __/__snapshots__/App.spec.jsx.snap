// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the main page should render Home component for root path 1`] = `
<div>
  Loading...
</div>
`;
